microsoft/testsuites/acctests/sgxtest.py

```diff
@@ -0,0 +1,217 @@
+        print("!!!!!!!!!!!!!!!!!!!!!!!!!!STARTED")
+        print(openenclave_install_path)
+        stdout = node.execute([openenclave_install_path + "bin/oesgx"], shell=True)
+        oesgx = stdout
+        sgx_version = ""
+        if "CPU supports Software Guard Extensions:SGX1" in oesgx:
+            sgx_version = "sgx1"
+        elif "CPU supports Software Guard Extensions:SGX2" in oesgx:
+            sgx_version = "sgx2"
+            
+    return sgx_version
+
+@staticmethod
+def isOeInstalled():
+    if not os.path.expanduser('~/openenclave-install/bin/'):
+        log.info("Not running test because openenclave binaries did not get installed.")
+        return False
+    return True
+
+
+def runOESamples(SAMPLES):
+    openenclave_install_path = getOEInstalledPath()
+    samples_stdout = ""
+    for sample in SAMPLES:
+        if sys.platform == "linux":
+            command = (
+                ". {}/share/openenclave/openenclaverc; make build; make run".format(
+                    openenclave_install_path
+                ),
+            )
+        elif sys.platform == "win32":
+            command = "vcvars64.bat & set OpenEnclave_DIR={}/lib/openenclave/cmake & cmake -G Ninja -DNUGET_PACKAGE_PATH=C:/oe_prereqs & ninja".format(
+                openenclave_install_path
+            )
+
+        returncode, stdout = execute(
+            command,
+            shell=True,
+            cwd="{}share/openenclave/samples/{}".format(
+                openenclave_install_path, sample
+            ),
+            env=my_env,
+        )
+        samples_stdout += stdout
+    return returncode
+
+#############################################################
+
+@TestSuiteMetadata(
+    area="ACC",
+    category="functional",
+    owner="acc@microsoft.com",
+    description="""
+    This is a part of ACC test suite.
+    This test case validates the installation of SGX. 
+    The Suite consists of 2 sets of tests:
+    1. High priority Level 0 test: verifyOE
+    2. Lower Priority Level 1 test: runAllOESamples 
+
+    verifyOE: Runs basic tests that validate SGX platform. 
+    runAllOESamples: Runs all samples
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class SgxTestSuite(TestSuite):
+    #This is a run_acc_tests.sh script initializer. 
+    #run_acc_tests.sh installs and sets up OE 
+    # On the node
+    #To Do: Remove unnecessary tests
+    def before_suite(self, log: Logger, **kwargs: Any) -> None:
+        self._acctest_script = CustomScriptBuilder(
+            Path(__file__).parent.joinpath("scripts"), ["run_acc_tests.sh"]
+        )
+    TIMEOUT = 2000
+    @TestCaseMetadata(
+        description="""
+        This test case validates functionality of Secure Guard Extensions platform.
+            1. Validate if OpenEnclave is installed on the Node
+            2. Get SGX version
+            3. Perform Remote Attestation and Hello world on the Node
+        """,
+        priority=0,
```

Comment:
The priority is too high. Only one provisioning test case is P0 so far. Please use P2 as default priority.